include/range/v3/view/bounded.hpp

```diff
@@ -89,12 +95,24 @@ namespace ranges
             struct bounded_fn
             {
                 template<typename Rng,
-                    CONCEPT_REQUIRES_(Range<Rng>() && !BoundedRange<Rng>())>
+                    CONCEPT_REQUIRES_(Range<Rng>() && !BoundedRange<Rng>() &&
+                        (!RandomAccessRange<Rng>() || !SizedRange<Rng>()))>
                 bounded_view<all_t<Rng>> operator()(Rng && rng) const
                 {
                     return bounded_view<all_t<Rng>>{all(std::forward<Rng>(rng))};
                 }
                 template<typename Rng,
+                    CONCEPT_REQUIRES_(Range<Rng>() && !BoundedRange<Rng>() &&
+                        RandomAccessRange<Rng>() && SizedRange<Rng>())>
+                iterator_range<range_iterator_t<all_t<Rng>>> operator()(Rng && rng) const
+                {
+                    auto first = ranges::begin(rng);
+                    auto last = first + ranges::distance(rng);
+                    return iterator_range<range_iterator_t<all_t<Rng>>>{
+                        detail::move(first), detail::move(last)
+                    };
+                }
```

Comment:
This overload does not participate in resolution when `Rng` is an rvalue non-`View`, since `all_t<Rng>` is ill-formed for rvalue non-`View` ranges.